RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducer.cc

```diff
@@ -8,55 +8,87 @@
 #include "FWCore/Utilities/interface/EDGetToken.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 #include "FWCore/Utilities/interface/StreamID.h"
-#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/SynchronizingEDProducer.h"
 #include "HeterogeneousCore/CUDACore/interface/JobConfigurationGPURecord.h"
 #include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
 #include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusterParamsDeviceCollection.h"
 #include "RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.h"
 #include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitTopologyRecord.h"
 
 namespace ALPAKA_ACCELERATOR_NAMESPACE {
-  class PFClusterSoAProducer : public stream::EDProducer<> {
+  class PFClusterSoAProducer : public stream::SynchronizingEDProducer<> {
   public:
     PFClusterSoAProducer(edm::ParameterSet const& config)
         : pfClusParamsToken(esConsumes(config.getParameter<edm::ESInputTag>("pfClusterParams"))),
           topologyToken_(esConsumes(config.getParameter<edm::ESInputTag>("topology"))),
           inputPFRecHitSoA_Token_{consumes(config.getParameter<edm::InputTag>("pfRecHits"))},
           outputPFClusterSoA_Token_{produces()},
           outputPFRHFractionSoA_Token_{produces()},
+          num_rhf_{cms::alpakatools::make_host_buffer<uint32_t>()},
           synchronise_(config.getParameter<bool>("synchronise")),
           pfRecHitFractionAllocation_(config.getParameter<int>("pfRecHitFractionAllocation")) {}
```

Comment:
Should `pfRecHitFractionAllocation_` (and the corresponding parameter in `fillDescriptions`) be removed as part of this PR ?

(for the backport to 14_1_X, this removal would trigger a new ConfDB parsing, unless `ParameterSetDescription::addOptional` is used in the backport PR, I think)

Reply:
It is certainly no longer a necessary, but I left it to be transparent to the menu. I'm happy to remove it here if you like?

Reply:
please do.

Reply:
@jsamudio, now that the parameter `pfRecHitFractionAllocation` has been removed, I would have been expecting a corresponding update in the `HLTrigger/Configuration/python/customizeHLTforCMSSW.py` file for this PR, such that the now unneeded parameters is removed from the existing HLT menus (where it currently exists).
If that's not done, we'll incur in this kind of exceptions:

```
----- Begin Fatal Exception 01-Oct-2024 08:57:42 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Validating configuration of module: class=PFClusterSoAProducer@alpaka label='hltParticleFlowClusterHBHESoA'
Exception Message:
Illegal parameter found in configuration.  The parameter is named:
 'pfRecHitFractionAllocation'
You could be trying to use a parameter name that is not
allowed for this plugin or it could be misspelled.
----- End Fatal Exception -------------------------------------------------
```

because the framework is trying to validate a parameter not available from the `fillDescriptions`.

Reply:
@mmusich Apologies, I will fix it shortly.

Reply:
Thank you @jsamudio,  no need to apologize :)

Reply:
I have:

```python
def customizeHLTfor46135(process):
    """Remove pfRecHitFractionAllocation from PFClusterSoAProducer config"""
    for producer in producers_by_type(process, "PFClusterSoAProducer@alpaka"):
        if hasattr(producer, 'pfRecHitFractionAllocation'):
            delattr(producer, 'pfRecHitFractionAllocation')
    for producer in producers_by_type(process, "alpaka_serial_sync::PFClusterSoAProducer"):
        if hasattr(producer, 'pfRecHitFractionAllocation'):
            delattr(producer, 'pfRecHitFractionAllocation')
    return process
```

I know this is only temporary, but is there a cleaner way to apply this change to the general `@alpaka` and explicit `serial_sync` producers in one step?

Reply:
I think this is fine,  we'll strive to reabsorb the customization quickly anyway.

Reply:
Actually, we will need to keep it for a while, to keep the v1.4 frozen menu working 🤷🏻‍♂️

Reply:
This customization was added to the PR.